tests/functional/context_methods/test_builtin_functions.py

```diff
@@ -55,6 +62,24 @@ def test_builtin_zip_function(self, project):
         assert f"zip_result: {expected_zip}" in log_output
         assert f"zip_strict_result: {expected_zip}" in log_output
 
+    def test_builtin_invocation_args_to_dict_function(self, project):
+        _, log_output = run_dbt_and_capture(
+            [
+                "--debug",
+                "run-operation",
+                "validate_invocation",
+                "--args",
+                "{my_variable: test_variable}",
+            ]
+        )
+        # Result is checked in two parts because profiles_dir is unique each test run
+
+        result = "invocation_result: {'debug': True, 'write_json': True, 'use_colors': True, 'printer_width': 80, 'version_check': True, 'partial_parse': True, 'static_parser': True, 'profiles_dir': "
+        assert result in log_output
+
+        second_result = "'send_anonymous_usage_stats': False, 'event_buffer_size': 100000, 'quiet': False, 'no_print': False, 'macro': 'validate_invocation', 'args': '{my_variable: test_variable}', 'which': 'run-operation', 'rpc_method': 'run-operation', 'indirect_selection': 'eager'}"
+        assert second_result in log_output
```

Comment:
This fails in our "structured logging" test, which has all log output in JSON format. Two options to get this passing:
- Add `--log-format`, `json` to the command arguments in `run_dbt_and_capture` above, and then adjust the expected `result` to account for `\` escapes when logs are JSON formatted
- Instead of logging out the dict, and then inspecting the log outputs, you could write the anticipated `result` dict in Jinja and perform a comparison there. Something like:
```
{% macro validate_invocation(my_variable) %}
    {% set expected_dict = {'debug': True, 'write_json': True, ...} %}
    {% if expected_dict != invocation_args_dict %}
        {% do exceptions.raise_compiler_error("Assertion failed") %}
    {% endif %}
{% endmacro %}
```

Reply:
Hey @jtcohen6 thanks for the feedback. I mentioned this same issue but on the PR rather than the code as I'm trying to work through it.

I think parsing the JSON logs is perhaps easier but makes the test harder to understand as it needs to iterate through a list of dict logs to find the matching string.

The issue with doing it in the macro is that the profiles_dir value in the invocation dict isn't static so it can't be a direct dictionary comparison

Reply:
'run_dbt_and_capture' replaces double blackslashes with an empty string, for easier log checking